orttraining/orttraining/core/graph/gradient_builder.cc

```diff
@@ -1836,5 +1836,21 @@ IMPLEMENT_GRADIENT_BUILDER(GetPadGradient) {
                               NodeDef("Pad", {GO(0), IA("Neg_pads")}, {GI(0)})};
 }
 
+IMPLEMENT_GRADIENT_BUILDER(GetScatterNDGradient) {
+  std::vector<NodeDef> result;
+  if (IsGradientRequiredForSrcNodeInput(0)) {
+    result.emplace_back(NodeDef("Shape", {I(2)}, {IA("Shape_updates")}));
+    result.emplace_back(NodeDef("ConstantOfShape", {IA("Shape_updates")}, {IA("Zero_Shape_updates")},
```

Comment:
Let's mark a potential optimization here, 
If the shape of I(2) is static, ConstantOfShape would be constant folded into an initializer, and potential resulting in a large tensor. 
Instead of materializing the entire tenor with zeros, we can use a compact tensor representation in this case.